src/Common/src/System/Net/Http/HttpHandlerDiagnosticListenerExtensions.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+    /// <summary>
+    /// Defines default values for http handler properties which is meant to be re-used across WinHttp and UnixHttp Handlers
+    /// </summary>
+    internal static class HttpHandlerDiagnosticListenerExtensions
+    {
+        public static Guid LogHttpRequest(this DiagnosticListener @this, HttpRequestMessage request)
+        {
+            if (@this.IsEnabled(HttpHandlerLoggingStrings.RequestWriteName))
+            {
+                Guid loggingRequestId = Guid.NewGuid();
+                long timestamp = Stopwatch.GetTimestamp();
+
+                @this.Write(
+                    HttpHandlerLoggingStrings.RequestWriteName,
+                    new
+                    {
+                        Request = request,
+                        LoggingRequestId = loggingRequestId,
+                        Timestamp = timestamp
+                    }
+                );
+
+                return loggingRequestId;
+            }
+
+            return Guid.Empty;
+        }
+
+        public static void LogHttpResponse(this DiagnosticListener @this, HttpResponseMessage response, Guid loggingRequestId)
+        {
+            if (@this.IsEnabled(HttpHandlerLoggingStrings.ResponseWriteName))
+            {
+                LogHttpResponseInner(@this, response, loggingRequestId);
+            }
+        }
+
+        public static void LogHttpResponse(this DiagnosticListener @this, Task<HttpResponseMessage> responseTask, Guid loggingRequestId)
+        {
+            if (@this.IsEnabled(HttpHandlerLoggingStrings.ResponseWriteName))
+            {
+                ScheduleLogResponse(@this, responseTask, loggingRequestId);
+            }
+        }
+
+        private static void ScheduleLogResponse(DiagnosticListener diagnosticListener, Task<HttpResponseMessage> responseTask, Guid loggingRequestId)
+        {
+            responseTask.ContinueWith(
+                t =>
+                {
+                    if (!t.IsFaulted &&
+                        !t.IsCanceled)
+                    {
+                        LogHttpResponseInner(diagnosticListener, t.Result, loggingRequestId);
+                    }
+                },
+                TaskScheduler.Default);
```

Comment:
My impression is that the answer is closer to "not very" than the alternative, but at the same time I don't have anything like a spec that says "perf isn't important."